src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1174,6 +1180,189 @@ class spell_pri_empowered_renew : public AuraScript
+    {
+        return ValidateSpellInfo({ SPELL_PRIEST_ENTROPIC_RIFT_AREATRIGGER });
+    }
+
+    void HandleApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* caster = GetCaster();
+        if (!caster)
+            return;
+
+        if (AreaTrigger* at = caster->GetAreaTrigger(SPELL_PRIEST_ENTROPIC_RIFT_AREATRIGGER))
+            at->SetDuration(GetDuration());
+    }
+
+    void Register() override
+    {
+        OnEffectApply += AuraEffectApplyFn(spell_pri_entropic_rift_aura::HandleApply, EFFECT_0, SPELL_AURA_MOD_SPEED_ALWAYS, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 447445 - Entropic Rift (AreaTrigger)
+struct areatrigger_pri_entropic_rift : public AreaTriggerAI
+{
+    using AreaTriggerAI::AreaTriggerAI;
+
+    void OnInitialize() override
+    {
+        _scheduler.Schedule(100ms, [this](TaskContext task)
+        {
+            UpdateMovement();
+            task.Repeat(250ms);
+        });
+    }
+
+    void OnCreate(Spell const* creatingSpell) override
+    {
+        Unit* caster = at->GetCaster();
+        if (!caster)
+            return;
+
+        caster->CastSpell(caster, SPELL_PRIEST_ENTROPIC_RIFT_AURA);
+
+        if (creatingSpell)
+        {
+            if (ObjectGuid const* targetGUID = std::any_cast<ObjectGuid>(&creatingSpell->m_customArg))
+                _targetGuid = *targetGUID;
+            _searchRadius = creatingSpell->GetSpellInfo()->GetMaxRange();
+        }
+    }
+
+    void OnUpdate(uint32 diff) override
+    {
+        _scheduler.Update(diff);
+
+        _damageTimer -= Milliseconds(diff);
+        while (_damageTimer <= 0s)
+        {
+            DealDamage();
+            _damageTimer += damagePeriod;
+        }
+    }
+
+    void UpdateMovement()
+    {
+        Unit* caster = at->GetCaster();
+        if (!caster)
+            return;
+
+        Unit* target = nullptr;
+        if (!_targetGuid.IsEmpty())
+            target = ObjectAccessor::GetUnit(*at, _targetGuid);
+
+        if (!target || !target->IsAlive())
+        {
+            target = FindNewEnemy(caster);
+            if (!target)
+            {
+                _targetGuid = ObjectGuid::Empty;
+                return;
+            }
+
+            _targetGuid = target->GetGUID();
+        }
+
+        PathGenerator path(at);
+        path.CalculatePath(target->GetPositionX(), target->GetPositionY(), target->GetPositionZ(), false);
+        at->InitSplines(path.GetPath());
+    }
+
+    void DealDamage()
+    {
+        Unit* caster = at->GetCaster();
+        if (!caster)
+            return;
+
+        float radius = at->GetMaxSearchRadius();
+        std::list<WorldObject*> targets;
+        Trinity::AnyUnfriendlyUnitInObjectRangeCheck checker(at, caster, radius);
+        Trinity::UnitListSearcher<Trinity::AnyUnfriendlyUnitInObjectRangeCheck> searcher(at, targets, checker);
+        Cell::VisitAllObjects(at, searcher, radius);
+
+        targets.remove_if([caster](WorldObject const* target)
```

Comment:
up to this day I don't understand why Trinity::AnyUnfriendlyUnitInObjectRange doesn't use Unit* as a baselist, especially if the worker itself explicits units, this would drop the need to cast to unit later